modules/admin/admin-ui/src/main/resources/web/admin/apps/content-studio/js/app/wizard/ContentWizardPanel.ts

```diff
@@ -845,6 +863,49 @@ export class ContentWizardPanel extends api.app.wizard.WizardPanel<Content> {
         }
     }
 
+    private doesFragmentContainId(fragmentPage: Page, id: ContentId): boolean {
+        let containsId = false;
+
+        if (fragmentPage) {
+            let fragmentCmp = fragmentPage.getFragment();
+            if (!!fragmentCmp && ObjectHelper.iFrameSafeInstanceOf(fragmentCmp.getType(), ImageComponentType)) {
+                containsId = (<ImageComponent>fragmentCmp).getImage().equals(id);
+            }
+        }
+        return containsId;
+    }
+
+    private doImageComponentsContainId(id: ContentId): wemQ.Promise<boolean> {
+        let page = this.getPersistedItem().getPage();
+        let fragments: ContentId[] = [];
+        let containsId = this.doRegionsContainId(page.getRegions().getRegions(), id, fragments);
+        if (!containsId && fragments.length > 0) {
+            return wemQ.all(fragments.map(fragmentId => new GetContentByIdRequest(fragmentId).sendAndParse()))
+                .then((fragmentContents: Content[]) => {
+                    return fragmentContents.some((fragmentContent: Content) => {
+                        return this.doesFragmentContainId(fragmentContent.getPage(), id);
+                    })
+                });
+        } else {
+            return wemQ(containsId);
+        }
+    }
+
+    private doRegionsContainId(regions: Region[], id: ContentId, fragments: ContentId[] = []): boolean {
+        return regions.some((region: Region) => {
+            return region.getComponents().some((component: Component) => {
+                if (ObjectHelper.iFrameSafeInstanceOf(component.getType(), FragmentComponentType)) {
+                    fragments.push((<FragmentComponent>component).getFragment());
+                    return false;
+                } else if (ObjectHelper.iFrameSafeInstanceOf(component.getType(), ImageComponentType)) {
+                    return (<ImageComponent>component).getImage().equals(id);
+                } else if (ObjectHelper.iFrameSafeInstanceOf(component.getType(), LayoutComponentType)) {
+                    return this.doRegionsContainId((<LayoutComponent>component).getRegions().getRegions(), id, fragments);
+                }
+            })
+        });
+    }
+
```

Comment:
Can you change this one so that it doesn't have ``else`` and has the last line with default ``return``?
```Typescript
{
  if (condition1) {
    do stuff;
    return false;
  }
  if (condition2) {
    return stuff;
  }
  return false;
}
```